src/Microsoft.AspNetCore.Mvc.Razor/RazorPageBase.cs

```diff
@@ -83,10 +83,10 @@ public virtual TextWriter Output
         public IDictionary<string, RenderAsyncDelegate> PreviousSectionWriters { get; set; }
 
         /// <summary>
-        /// Gets or sets a <see cref="System.Diagnostics.DiagnosticSource"/> instance used to instrument the page execution.
+        /// Gets or sets a <see cref="System.Diagnostics.DiagnosticListener"/> instance used to instrument the page execution.
         /// </summary>
         [RazorInject]
-        public DiagnosticSource DiagnosticSource { get; set; }
+        public DiagnosticListener DiagnosticSource { get; set; }
```

Comment:
>public DiagnosticListener DiagnosticSource { get; set; } [](start = 8, length = 56)

Would it be possible to achieve the same effect using a private/protected internal field of the desired type? I'm pretty worried about binary compat here.

I'd like to try and understand what sort of perf improvments we can make by removing the instrumentation totally.

If we make breaking changes, we want the benefit to be significant, so we might want to go all of the way.